lib/sailor.js

```diff
@@ -322,29 +346,118 @@ class Sailor {
                 const headers = _.clone(outgoingMessageHeaders);
                 headers.messageId = data.id || headers.messageId;
                 logger.trace({
-                    messagesCount: self.messagesCount,
+                    messagesCount: that.messagesCount,
                     messageProcessingTime: Date.now() - timeStart
                 }, 'processMessage emit data');
 
                 headers.end = new Date().getTime();
 
                 if (stepData.is_passthrough === true) {
+                    data.passthrough = { ...origPassthrough };
                     if (settings.NO_SELF_PASSTRHOUGH) {
                         const { stepId } = incomingMessageHeaders;
                         if (stepId) {
                             data.passthrough = Object.assign({}, origPassthrough, {
                                 [stepId]: Object.assign({}, _.omit(payload, 'passthrough'))
                             });
                         }
+                    }
+                }
+
+                data.headers = data.headers || {};
+                const { body, passthrough } = data;
+
+                if (settings.EMIT_LIGHTWEIGHT_MESSAGE) {
```

Comment:
Is it possible to cut new code into smaller functions?
It's quite complicated already want to simplify

Reply:
i've already done what i can with moving logic out of processMessage into runExec

Reply:
futher steps require heavy refactoring of sailor